csharp/src/Drivers/BigQuery/BigQueryStatement.cs

```diff
@@ -273,7 +273,10 @@ private string SerializeToJson(StructArray structArray, int index)
                     }
                     else
                     {
-                        jsonDictionary.Add(name, ParseStructArray(structArray1, index));
+                        if (index < structArray1.Length)
```

Comment:
The logic in this method might be easier to follow with a few strategic comments distinguishing between the various branches it's possible to take.